src/main/scala/com/cloudera/sparkts/DateTimeIndex.scala

```diff
@@ -115,30 +115,32 @@ trait DateTimeIndex extends Serializable {
  * for constant space storage and constant time operations.
  */
 class UniformDateTimeIndex(
-    val start: Long,
+    val start: ZonedDateTime,
     val periods: Int,
     val frequency: Frequency,
-    val dateTimeZone: DateTimeZone = DateTimeZone.getDefault())
+    val dateTimeZone: ZoneId = ZoneId.systemDefault())
   extends DateTimeIndex {
 
-  override def first: DateTime = new DateTime(start, dateTimeZone)
+  override def first: ZonedDateTime = start
 
-  override def last: DateTime = frequency.advance(new DateTime(first, dateTimeZone), periods - 1)
+  override def last: ZonedDateTime = frequency.advance(first, periods - 1)
 
-  override def zone: DateTimeZone = dateTimeZone
+  override def zone: ZoneId = dateTimeZone
 
   override def size: Int = periods
 
   override def slice(interval: Interval): UniformDateTimeIndex = {
-    slice(interval.start, interval.end)
+    slice(ZonedDateTime.ofInstant(interval.getStart(), zone),
+          ZonedDateTime.ofInstant(interval.getEnd(), zone))
```

Comment:
This should be indented back to two spaces past the line above.  This same comment applies to a bunch of other places in the patch.